### PR TITLE
Add remote port option to the HTTP access log formats

### PR DESCRIPTION
**Pull request by kgibm**

### CODE REVIEW

**Comment by kgibm** [comment]
#build

File: dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/channel/internal/values/AccessLogRemotePort.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        HttpRequestMessageImpl requestMessageImpl = null;
+        String remotePort = null;
+        if (request != null) {
+            requestMessageImpl = (HttpRequestMessageImpl) request;
+        }
+
+        if (requestMessageImpl != null) {
+            remotePort = Integer.toString(requestMessageImpl.getServiceContext().getRemotePort());
+        }
+        return remotePort;
```
Comment (iBaminian) [inline dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/channel/internal/values/AccessLogRemotePort.java:44]: If th requestMessageImpl happened to be null, then the directive would return null on remotePort. I believe other directives will return the dash "-" character

File: dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/channel/internal/values/AccessLogRemotePort.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
+/*******************************************************************************
+ * Copyright (c) 2004, 2008 IBM Corporation and others.
```
Comment (mrsaldana) [inline dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/channel/internal/values/AccessLogRemotePort.java]: This should probably range up to 2021

**Comment by kgibm** [comment]
The following test failure occurs in CL in `dev/com.ibm.ws.transport.http_test/publish/files/accessLogging/server-invalid.xml` because `%e` is used as a test for an invalid log format so that would need to be changed.

```
Caused by: junit.framework.ComparisonFailure: Wrong line expected:<%z [%{test}e] %{> but was:<%z [49628] %{>
	at test.server.transport.http.AccessLogTest.testAccessLoggingInvalid(AccessLogTest.java:360)
```

**Comment by kgibm** [comment]
#build

**Comment by kgibm** [comment]
fixes #18132

**Comment by kgibm** [comment]
Uses `%{remote}p` to match [httpd](https://httpd.apache.org/docs/trunk/mod/mod_log_config.html#formats). Example configuration with both `%{remote}p` and `%p` as the last two elements:

```
<accessLogging filepath="${server.output.dir}/logs/http_access.log" maxFileSize="100" maxFiles="4" logFormat="%h %u %t &quot;%r&quot; %s %b %D %{R}W %{remote}p %p" />
```

Example output of a `curl` to Liberty on port `9080`:

```
127.0.0.1 - [16/Aug/2021:10:42:24 -0700] "GET /swat/ HTTP/1.1" 200 21983 5625 3708 59212 9080
```

Remote ephemeral port `59212` matches Wireshark trace:

wireshark

**Comment by kgibm** [comment]
#build

**Comment by kgibm** [comment]
#build

**Comment by kgibm** [comment]
There were a few failures in `com.ibm.ws.ui.tool.serverConfig_fat` that don't seem to be related.

**Review by mrsaldana — APPROVED** [review]
Changes look good, thanks for this work!

File: dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/channel/internal/values/AccessLogRemotePort.java (hunk 2 above)
Comment (kgibm) [reply dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/channel/internal/values/AccessLogRemotePort.java]: Combined AccessLogRemotePort and AccessLogLocalPort into AccessLogPort and removed AccessLogRemotePort

**Comment by kgibm** [comment]
#run-libby-bot

**Comment by kgibm** [comment]
Rebased on latest release branch

#build

**Comment by kgibm** [comment]
Add beta fence

#build

**Comment by kgibm** [comment]
Support beta fence in more FAT tests

#build

**Comment by kgibm** [comment]
#build

**Comment by kgibm** [comment]
#run-libby-bot

**Comment by kgibm** [comment]
Latest build looks good. Three FAT buckets had errors but they all seemed to be unrelated to logging (e.g. a deadlock in JBatch). Downloaded the logs of all three to double check for the beta fence error and didn't find it.

**Comment by kgibm** [comment]
#run-libby-bot

**Pull request merged**